data_processing.tex

```diff
@@ -5,27 +5,29 @@ \section{Data Processing}
-    generators which keeps resource use efficient.
+  	Sentence preprocessing was conducted in order to clean the dataset prior to being fed into classifiers.
+  	Preprocessing steps include stemming, lemmatisation and stopword removal. These different options are made available separately so that classifiers can use them as required.
 
   \subsection{Word2Vec}
   \label{sec:word2vec}
-  Words must be embedded into a numerical space for the majority of learning methods. Word2Vec uses a two-layer neural network to autoencode the words against their neighbours in the input corpus. Word2Vec produces word encodings in large vector-spaces. Semantic analogies are preserved in the space, (i.e. $queen - king + man = woman$). We trained Word2Vec on our own corpus to embed words for most of our algorithms, however, for our windowed deep neural network, we used the `Global Vectors for Word Representation` (GloVe) algorithm using a pre-trained vector of 6B words from wikipedia. These two methods are compared in Table \ref{tab:window_res}.
+  One-hot encoding of words leads to impractically large dimensionality, and it is desirable to instead create a lower dimensional embedding for the vocabulary being used.
+  
+  Word2Vec uses a two-layer neural network to autoencode words using their neighbours in the input corpus. The resulting space is of far lower dimensionality, and preserves semantic relationships, (i.e. $queen - king + man = woman$). A Word2Vec embedding was trained on our own corpus. However due to the relatively small dataset, it is likely that this embedding does not strongly enforce relationships between words and will not contain words that are present in the test data.
+  
+To address this, a pre-trained encoding was also used: A `Global Vectors for Word Representation` (GloVe) representation trained on 6 billion words from wikipedia.
 
   \subsection{Sentence Transformations}
   \label{sec:sentence_transformations}
 
-    It is necessary to transform the variable-length sentences to a fixed length
-    representation. Two methods have been used throughout the project: padding
-    (or truncating) sentences to a known length using a special padding symbol and
-    producing windows into a sentence. Several windows can be gained from each
+    For many learning machines, inputs of fixed length
+    are required. Two methods have been used throughout this project: padding
+    (or truncating) sentences to a known length and
+    windowing. Several windows can be gained from each
     sentence: augmenting the dataset.
 
-    For example, the sentence \textit{``This process, however, afforded me no
-      means of ascertaining the dimensions of my dungeon [...]''} would produce windows
-    such as \texttt{[this process , however ,], [process ,
-      however , afforded], [, however , afforded me], ...}
-
     Two encodings for the class labels have been experimented with: one hot
```

Comment:
lines 28 and 29 need to be removed as they are now superseeded by the new author representation subsection